File: spec/java/src/io/kaitai/struct/specwrite/TestEosExceptionBytes.java
package io.kaitai.struct.specwrite;

import io.kaitai.struct.ByteBufferKaitaiStream;
import io.kaitai.struct.KaitaiStream;
import io.kaitai.struct.KaitaiStruct.ReadWrite;
import io.kaitai.struct.testwrite.EosExceptionBytes;
import org.testng.annotations.Test;
import org.testng.SkipException;

public class TestEosExceptionBytes extends CommonSpec {
    @Override
    protected Class<? extends ReadWrite> getStructClass() {
        return EosExceptionBytes.class;
    }

    @Override
    protected String getSrcFilename() {
        return "term_strz.bin";
    }

    @Override
    @Test
    protected void testReadWriteRoundtrip() throws Exception {
        throw new SkipException("cannot use roundtrip because parsing is expected to fail");
    }

    @Test(expectedExceptions = java.nio.BufferOverflowException.class)
    public void testEosExceptionBytes() throws Exception {
        EosExceptionBytes r = new EosExceptionBytes();

        EosExceptionBytes.Data data = new EosExceptionBytes.Data(null, r, r._root());
        data.setBuf(new byte[] { 0, -1, -2, -3, -4, -5, -6 });
        data._check();

        r.setEnvelope(data);
        r._check();

        try (KaitaiStream io = new ByteBufferKaitaiStream(12)) {
            r._write(io);
        }
    }
}

File: spec/python/specwrite/test_eos_exception_bytes.py
import unittest
import io
from kaitaistruct import KaitaiStream
from specwrite.common_spec import CommonSpec

from testwrite.eos_exception_bytes import EosExceptionBytes

class TestEosExceptionBytes(CommonSpec.Base):
    def __init__(self, *args, **kwargs):
        super(TestEosExceptionBytes, self).__init__(*args, **kwargs)
        self.struct_class = EosExceptionBytes
        self.src_filename = 'src/term_strz.bin'

    def test_read_write_roundtrip(self):
        self.skipTest("cannot use roundtrip because parsing is expected to fail")

    def test_eos_exception_bytes(self):
        r = EosExceptionBytes()

        data = EosExceptionBytes.Data(None, r, r._root)
        data.buf = b"\x00\xFF\xFE\xFD\xFC\xFB\xFA"
        data._check()

        r.envelope = data
        r._check()

        with KaitaiStream(io.BytesIO(bytearray(12))) as out_io:
            with self.assertRaisesRegex(EOFError, u"^requested to write 7 bytes, but only 6 bytes left in the stream$"):
                r._write(out_io)
